test/unit-test/test_lib/test_vis.py

```diff
@@ -29,7 +32,20 @@ def test_calc_plot_linear_fit(self, x_in, y_in, expected_result):
         os.remove(current_dir + '/linear_plot_test.png')
         
         assert np.allclose(output, expected_result)
-    
+
+    @patch("matplotlib.pyplot.plot")
+    @patch("matplotlib.pyplot.savefig")
+    def test_plot_data(self,mock_savefig,mock_plot):
+        data = ([[0,1,2,3,4,5,6,7,8],[0,2,4,6,8,10,12,14,16]])
+        x_data, y_data = (d for d in data)
```

Comment:
```
(d for d in data)
```
Is the above coding idiom common in Python? @bhuvangupta008 If you can show an example from a respected project / coder, then I would go along with it.

Reply:
I used it in conformation with the [vis.py](https://github.com/bhuvangupta008/IRCLogParser/blob/bhuvan-test/lib/vis.py#L26) file as used in the project. 
Alternative would have been, accessing the tuple members via index like 
```
x_data = data[0]
y_data = data[1]
```